File: python/android_local_test.py
# coding=utf-8
import unittest
from appium import webdriver


class AndroidLocalTest(unittest.TestCase):
    def setUp(self):
        self.driver = None

    def tearDown(self):
        self.driver.quit()

    def test_chrome(self):
        # 需要下载chrome，并已安装
        web_desired_caps = {
            'platformName': 'Android',
            'platformVersion': '7.0',
            'deviceName': '520381b347dd148b',
            'browserName': 'Chrome',  # 想启动内置浏览器则将值改为"Browser"即可
        }
        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', web_desired_caps)
        self.driver.get("https://www.baidu.com/")
        self.assertEqual(self.driver.current_url, "https://www.baidu.com/")
        self.assertEqual(self.driver.title, u"百度一下")

    def test_calculator(self):
        calculator_desired_caps = {
            'platformName': 'Android',
            'platformVersion': '7.0',
            'deviceName': '520381b347dd148b',
            # 如果想要运行其他的内置APP，则将package、activity替换即可
            # 获取计算器的package、activity即可
            # 可通过 adb shell dumpsys window windows | findstr "Current" 获取，（首先打开计算器，然后运行命令即可获取到）
            'appPackage': 'com.sec.android.app.popupcalculator',
            'appActivity': 'com.sec.android.app.popupcalculator.Calculator'
        }
        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', calculator_desired_caps)
        self.driver.find_element_by_accessibility_id("5").click()
        self.driver.find_element_by_accessibility_id("Plus").click()
        self.driver.find_element_by_accessibility_id("6").click()
        self.driver.find_element_by_accessibility_id("Equal").click()
        self.assertEqual(self.driver.find_element_by_id("txtCalc").text, "11")


if __name__ == '__main__':
    suite = unittest.TestLoader().loadTestsFromTestCase(AndroidLocalTest)
    unittest.TextTestRunner(verbosity=2).run(suite)
    # unittest.main()

File: samples/java/src/test/java/AndroidLocalTest.java
import org.apache.logging.log4j.Logger;
import org.apache.logging.log4j.LogManager;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import io.appium.java_client.android.AndroidDriver;

import java.net.MalformedURLException;
import java.net.URL;

public class AndroidLocalTest {
    private static AndroidDriver<?> driver;
    private static Logger log= LogManager.getLogger("appium-log");

    @BeforeTest
    public void beforeTest( ) {
        log.info("######################### begin test #########################");
    }

    @AfterTest
    public void afterTest( ) {
        driver.quit();
    }

    @Test
    public void localCalculator() throws MalformedURLException{
        DesiredCapabilities cap = new DesiredCapabilities();
        cap.setCapability("platformName","'Android'");
        cap.setCapability("platformVersion","7.0");
        cap.setCapability("deviceName","520381b347dd148b");
        // 获取计算器的package、activity即可
        // （可通过adb shell dumpsys window windows | findstr "Current"获取，首先打开计算器，然后运行命令即可获取到）
        // 如果想要运行其他的内置APP，则将package、activity替换即可
        cap.setCapability("appPackage","com.sec.android.app.popupcalculator");
        cap.setCapability("appActivity","com.sec.android.app.popupcalculator.Calculator");
        driver = new AndroidDriver<>(new URL("http://127.0.0.1:4723/wd/hub"), cap);
        driver.findElementByAccessibilityId("5").click();
        driver.findElementByAccessibilityId("Plus").click();
        driver.findElementByAccessibilityId("6").click();
        driver.findElementByAccessibilityId("Equal").click();
        Assert.assertEquals(driver.findElementById("txtCalc").getText(),"11");
    }

    @Test
    public void launchChrome() throws MalformedURLException {
        //需要下载chrome，并已安装
        DesiredCapabilities cap = new DesiredCapabilities();
        cap.setCapability("platformName","'Android'");
        cap.setCapability("platformVersion","7.0");
        cap.setCapability("deviceName","520381b347dd148b");
        // 想启动内置浏览器则将值改为"Browser"即可
        cap.setCapability("browserName","chrome");
        // 该步骤和下边注释的两条任意选择一个都可启动
        //chrome -- com.android.chrome/com.google.android.apps.chrome.Main
//            cap.setCapability("appPackage","com.android.chrome");
//            cap.setCapability("appActivity","com.google.android.apps.chrome.Main");
        driver = new AndroidDriver<>(new URL("http://127.0.0.1:4723/wd/hub"), cap);
        driver.get("https://www.baidu.com/");
        log.info("############################"+driver.getCurrentUrl()+"##########################");
        log.info("############################"+driver.getTitle()+"##########################");
        Assert.assertEquals(driver.getCurrentUrl(), "https://www.baidu.com/", "URL Mismatch");
        Assert.assertEquals(driver.getTitle(), "百度一下", "Title Mismatch");
    }
}
